libyul/backends/wasm/polyfill/Comparison.yul

```diff
@@ -35,15 +35,12 @@ function iszero512(x1, x2, x3, x4, x5, x6, x7, x8) -> r:i32 {
 }
 
 function eq(x1, x2, x3, x4, y1, y2, y3, y4) -> r1, r2, r3, r4 {
-	if i64.eq(x1, y1) {
-		if i64.eq(x2, y2) {
-			if i64.eq(x3, y3) {
-				if i64.eq(x4, y4) {
-					r4 := 1
-				}
-			}
-		}
-	}
+//	r4 := i32.and(i64.eq(x1, y1), i32.and(i64.eq(x2, y2), i32.and(i64.eq(x3, y3), i64.eq(x4, y4))))
```

Comment:
This is a branch-less version, and the one below is using leave. Both are better than the nested conditions in my opinion.

@aarlt @chriseth which one is nicer?

Reply:
I agree with you, both versions are better than the nested if's. I'm not really sure what version is nicer. Maybe the version using `leave` is easier to read.

Reply:
Functions containing `leave` can currently not be inlined.

Reply:
So for potentially inlineable functions, should we prefer nested branches or branch-less versions as above? (Since leave is not preferred.)

Reply:
I think the branch-less version should behave best.